src/DatePicker.js

```diff
@@ -1,41 +1,29 @@
-import React from 'react';
+import React, { useMemo, useEffect, useRef } from 'react';
 import PropTypes from 'prop-types';
 import TextInput from './TextInput';
 import idgen from './idgen';
 
-class DatePicker extends React.Component {
-  constructor(props) {
-    super(props);
+const DatePicker = props => {
+  const id = useMemo(() => props.id || `datepicker${idgen()}`, [props.id]);
+  const instance = useRef(null);
 
-    this.id = props.id || `datepicker${idgen()}`;
-  }
-
-  componentDidMount() {
-    if (typeof M !== 'undefined') {
-      const { onChange } = this.props;
-      const elem = document.getElementById(this.id);
-      const options = onChange
-        ? { ...this.props.options, onSelect: onChange }
-        : this.props.options;
-
-      this.instance = M.Datepicker.init(elem, options);
-    }
-  }
+  useEffect(() => {
+    const { onChange } = props;
+    const options = onChange
+      ? { ...props.options, onSelect: onChange }
+      : props.options;
+    const elem = document.getElementById(id);
+    instance.current = M.Datepicker.init(elem, options);
```

Comment:
Also something else i've seen :D no need to create a ref since the instance is only used inside this scope, so `const instance = M...` is enough